src/firefly/js/ui/panel/TabPanel.jsx

```diff
@@ -34,14 +100,13 @@ export class Tabs extends Component {
     }
 
     componentWillReceiveProps(nextProps) {
-        this.state= tabsStateFromProps(nextProps);
+        this.state= Object.assign(tabsStateFromProps(nextProps));
```

Comment:
what does this 'assign' statement do?  for assign, don't you need at least 2 parameters?